frontend/src/components/Cards/CardStore.js

```diff
@@ -2,105 +2,126 @@ import React from 'react';
+  }
 
+  determineSelectedCard() {
     let selected = this.state.selectedCard;
 
     if (selected && selected.length > 0) {
       const messageObject = {
-        from: this.props.currentUser,
+        user: this.state.username,
+        room: this.state.gameRoom,
         data: selected
       };
 
-      GameInstance.sendSelectedCard(messageObject);
+      this.sendSelectedCard(messageObject);
       this.setState({
-        message: ''
-      })
+        message: ""
+      });
     }
   }
 
-    handleChange(e) {
-        this.setState({ value: e });
-        this.props.callbackSendCardSelectionOut(e);
-    }
+  sendSelectedCard(envelope) {
+    GameInstance.sendMessage({
+      command: "send_cards_request",
+      user: envelope.user,
+      room: envelope.room,
+      payload: envelope.data
+    });
+  }
+
+  handleChange(e) {
+    this.setState({ value: e });
+    this.props.callbackSendCardSelectionOut(e);
+  }
+
+  clearSelected(e) {
+    this.setState({ value: [] });
+  }
+
+  async selectCard() {
+    try {
+      this.determineSelectedCard();
+      let result = await this.requestCard();
 
-    clearSelected(e) {
-        this.setState({ value: [] });
+      if (result) {
+        this.setState({ selectedCard: result });
+      }
+    } catch (exception) {
+      console.log(exception);
     }
+  }
 
+  async requestCard() {
+    try {
 
-    render() {
-        return (
-            <container className="card">
-            <div className='row'>
-             <div className='col-sm'>
-            <div id='card'>
-            <Card>
-                <Card.Body>
-                <Card.Title>{this.state.cardName}</Card.Title>
-                 <Card.Subtitle className="mb-2 text-muted">{this.state.cardCost}</Card.Subtitle>
-                    <Card.Text>{this.state.cardType}</Card.Text>
-                    <Card.Text>{this.state.cardEffect}</Card.Text>
-                    <Card.Text>{this.state.cardFootnote}</Card.Text>
-                    </Card.Body>
-            </Card>
-
-            <Card>
-                <Card.Body>
-                <Card.Title>{this.state.cardName}</Card.Title>
-                 <Card.Subtitle className="mb-2 text-muted">{this.state.cardCost}</Card.Subtitle>
-                    <Card.Text>{this.state.cardType}</Card.Text>
-                    <Card.Text>{this.state.cardEffect}</Card.Text>
-                    <Card.Text>{this.state.cardFootnote}</Card.Text>
-                    </Card.Body>
-            </Card>
-
-            <Card>
-                <Card.Body>
-                <Card.Title>{this.state.cardName}</Card.Title>
-                 <Card.Subtitle className="mb-2 text-muted">{this.state.cardCost}</Card.Subtitle>
-                    <Card.Text>{this.state.cardType}</Card.Text>
-                    <Card.Text>{this.state.cardEffect}</Card.Text>
-                    <Card.Text>{this.state.cardFootnote}</Card.Text>
+      const res = await fetch(Constants.REST_ENDPOINT_CARD);
```

Comment:
No need for REST here.. See consumers.py get_the_three_store_cards comment below... you want to send the cards back to the client.

Reply:
Removed REST point as mentioned